hack/ci/create_devstack.sh

```diff
@@ -89,7 +93,11 @@ function ensure_openstack_client {
 
         # Install virtualenv to install the openstack client and curl to fetch
         # the build constraints.
-        apt-get install -y python3-virtualenv curl
+        if [[ $UID -ne 0 ]]; then
+            sudo apt-get install -y python3-virtualenv curl
```

Comment:
`root` should also can sudo in ubuntu ?

Reply:
Usually, sure. But [the Prow test image](https://github.com/kubernetes/test-infra/blob/1757d0d69fbc9298732980fdc966c54717f27d08/config/jobs/kubernetes-sigs/cluster-api-provider-openstack/cluster-api-provider-openstack-presubmits.yaml#LL60C55-L60C55) does not have `sudo` ([see this failed e2e job](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1539/pull-cluster-api-provider-openstack-e2e-test/1655723663397228544) and this https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1539#issuecomment-1539720740).